File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/gui/FrmImportarDiretorioProgresso.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 08/07/2015
 * Time: 11:26
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using HFSGuardaDiretorio.comum;

namespace HFSGuardaDiretorio.gui
{
	/// <summary>
	/// Description of FrmImportarDiretorioProgresso.
	/// </summary>
	public class FrmImportarDiretorioProgresso : IProgressoLog
	{
	    FrmImportarDiretorio form;

	    public FrmImportarDiretorioProgresso(FrmImportarDiretorio form) 
	    {
	        this.form = form;
		}
	    
		public void ProgressoLog(Progresso progresso)
		{
	        if (progresso.Posicao == 0) {
	            this.form.pbImportar.Minimum = progresso.Minimo;
	            this.form.pbImportar.Maximum = progresso.Maximo;
	            this.form.pbImportar.Step = progresso.Passo;
	        }
	        this.form.pbImportar.Value = progresso.Posicao;	        
	        
	        if (progresso.Log.Length > 0) {
	            int pos = form.memoImportaDir.TextLength;
	            form.memoImportaDir.AppendText(progresso.Log+"\n");
	            form.memoImportaDir.Select(pos, pos);
	            form.barraStatus.Items[1].Text = Convert.ToString(pos);
	            form.barraStatus.Update();
	        }
	        
		}
	    
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/gui/FrmImportarDiretorioProgresso.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 08/07/2015
' * Time: 11:26
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports HFSGuardaDiretorio.comum

Namespace gui
	''' <summary>
	''' Description of FrmImportarDiretorioProgresso.
	''' </summary>
	Public Class FrmImportarDiretorioProgresso
		Implements IProgressoLog
		Private form As FrmImportarDiretorio

		Public Sub New(form As FrmImportarDiretorio)
			Me.form = form
		End Sub

		Public Sub ProgressoLog(progresso As Progresso) Implements IProgressoLog.ProgressoLog
			If progresso.Posicao = 0 Then
				Me.form.pbImportar.Minimum = progresso.Minimo
				Me.form.pbImportar.Maximum = progresso.Maximo
				Me.form.pbImportar.[Step] = progresso.Passo
			End If
			Me.form.pbImportar.Value = progresso.Posicao

			If progresso.Log.Length > 0 Then
				Dim pos As Integer = form.memoImportaDir.TextLength
				form.memoImportaDir.AppendText(progresso.Log & vbLf)
				form.memoImportaDir.[Select](pos, pos)
				form.barraStatus.Items(1).Text = Convert.ToString(pos)
				form.barraStatus.Update()
			End If

		End Sub

	End Class
End Namespace
